### ISSUE TITLE
[SOLVED] Console Startup Errors

### ISSUE THREAD

**Issue by x0Z3ro0x** [ESX-Org/es_extended#98]
Thank you for taking the time to read this. I love your work but I think I screwed up somewhere. Here are my errors:

https://pastebin.com/G2JNbMcy

I am sure this is my own doing. Any and all help is greatly appreciated. I have been unable to contact you any other way, so I apologize if using your git page offends you.

I am using the latest version of your plugin, on Windows Server 2016. Anything you need just ask I will gladly supply it. Thank you again and I hope to hear from you soon.

**Comment by x0Z3ro0x** [comment]
I solved these errors through brutal searching, and snotty replies, but I still did it! So to those that stumble upon this, here was my fix and I hope it can help you a lot sooner than it helped me and with a lot less shitty attitudes:

The convar `set mysql_connection_string` must be ABOVE the `start [scripts]` section of your server.cfg despite what FiveM.net has you believe and follow. In addition you must add some lines to that convar changing the standard:

`set mysql_connection_string "server=127.0.0.1;database=essentialmode;userid=root;password=changedforgithub`

to

`set mysql_connection_string "server=127.0.0.1;database=essentialmode;userid=root;password=Systemkiller.45;sslmode=none;AllowPublicKeyRetrieval=true"`

Why this worked I do not know I am not that far along yet. What I do know is it worked for me and may just work for someone else. These changes made the world of difference for our server and allowed us to soar ahead to new errors to conquer :D As long as you are adding you will continue to get new errors, but fixing them and moving on is always a good thing because it means more features and a better server for your users.

**Comment by x0Z3ro0x** [comment]
I was able to solve my error through very hard work and searching. I had to deal with several rude people to get here. So I am posting my fix in hopes that the next users stumbles upon it and can solve their issue faster and with less headache then I did.

The error above and future errors once you solve those errors above can be corrected all at once by doing the following:

First despite what FiveM.net has you follow, you need to cut and paste your `set convar` section ABOVE your `start [scripts]` section in your `server.cfg`. This is incredibly important for future errors.

Next we need to change our convar `set mysql_connection_string` from it's default of:

`set mysql_connection_string "server=127.0.0.1;database=essentialmode;userid=root;password=changedforgithub"`

to

`set mysql_connection_string "server=127.0.0.1;database=essentialmode;userid=root;password=changedforgithub;sslmode=none;AllowPublicKeyRetrieval=true"`

Why this works I do not know. That is still beyond my knowledge at the moment, however what I do know is that it works. The changes are adding `sslmode=none` and `AllowPublicKeyRetrieval=true` to your convar `set mysql_connection_string`. It allowed me to get nearly every feature on the server working and start storing / loading info from the DB properly. Since then I have new errors to solve and once I do, I will post the solutions to those as well. Together, we will build our servers!

**Comment by ElPumpo** [comment]
Quick googling: https://forum.fivem.net/t/mysql-errors-and-a-couple-others-help-needed/125190/2, seems like something isn't properly installed - I'd just reinstall Windows, or whatever OS you're using. x509 issues are not fun, I know.

**Comment by x0Z3ro0x** [comment]
That's my thread installed. The users in there specifically d0p3t didn't know what they were talking about. In fact I followed their suggestions and uninstalled my windows server, deleted my first attempt at creating a server, installed linux and ending up with MORE errors than I had before. So I reinstalled windows (still had the same errors.

No one had a clue why the errors were happening. The fix I posted above is what solved the errors. Turns out everything was installed just fine the problem was the server.cfg for me.
